### ISSUE TITLE
EOS Token Handling in ParlerTTS Loss Function

### ISSUE THREAD

**Issue by shamuiscoding** [huggingface/parler-tts#208]
### [Description]
While reviewing the training code of `ParlerTTSForCausalLM`, I noticed a potential issue regarding the handling of `eos_token_id`. The code appears to ignore all `eos_token_ids` during loss calculation, despite the comment stating:

"we use every codebook's token AND one single EOS at the end of each codebook"

The logic in `ParlerTTSLogitsProcessor` expects an EOS token to be generated at the first codebook. Since `eos_token_ids` are fully ignored during training, wouldn't this make it impossible for the model to predict EOS tokens correctly at inference time?

### [Code Reference]
in `modeling_parler_tts.py`
```
  # (bsz, vocab_size, seq_len, num_codebooks), (bsz, seq_len, num_codebooks)
  labels = labels.masked_fill(labels == self.config.bos_token_id, -100)

  # we use every codebooks token AND one single EOS at the end of each codebooks
  mask = (input_ids.transpose(1, 2) != self.config.eos_token_id) & ((labels != -100))
```

-----

Would appreciate any clarification on whether this is intentional or a potential issue.
